roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md

```diff
@@ -4,10 +4,324 @@ The coordinator is the central subsystem of the node-side components which parti
+  is found during approval checking.
+- Ensuring malicious approval votes will be recorded, so nodes can get slashed
+  properly.
+- Coordinating actual participation in a dispute, ensuring that the node
+  participates in any justified dispute in a way that ensures resolution of
+  disputes on the network even in the case of many disputes raised (flood/DoS
+  scenario).
+- Provide an API for chain selection, so we can prevent finalization of any
+  chain which has included candidates for which a dispute is either ongoing or
+  concluded invalid and avoid building on chains with an included invalid
+  candidate.
+- Provide an API for retrieving (resolved) disputes, including all votes, both
+  implicit (approval, backing) and explicit dispute votes. So validators can get
+  rewarded/slashed accordingly.
+
+## Ensuring That Disputes Can Be Raised
+
+If a candidate turns out invalid in approval checking, the `approval-voting`
+subsystem will try to issue a dispute. For this, it will send a message
+`DisputeCoordinatorMessage::IssueLocalStatement` to the dispute coordinator,
+indicating to cast an explicit invalid vote. It is the responsibility of the
+dispute coordinator on reception of such a message to create and sign that
+explicit invalid vote and  trigger a dispute if none is already
+ongoing.
+
+In order to raise a dispute, a node has to be able to provide two opposing votes.
+Given that the reason of the backing phase is to have validators with skin in
+the game, the opposing valid vote will very likely be a backing vote. It could
+also be some already cast approval vote, but the significant point here is: As
+long as we have backing votes available, any node will be able to raise a
+dispute.
+
+Therefore a vital responsibility of the dispute coordinator is to make sure backing
+votes are available for all candidates that might still get disputed. To
+accomplish this task in an efficient way the dispute-coordinator relies on chain
+scraping. Whenever a candidate gets backed on chain, we record in
+chain storage the backing votes (gets overridden on every block). We provide a
+runtime API for querying those votes. The dispute coordinator makes sure to
+query those votes for any non finalized blocks: In case of missed blocks, it
+will do chain traversal as necessary.
+
+Relying on chain scraping is very efficient for two reasons:
+
+1. Votes are already batched. We import all available backing votes for a
+   candidate all at once. If instead we imported votes from candidate-backing as
+   they came along, we would import each vote individually which is
+   inefficient in the current dispute coordinator implementation (quadratic
+   complexity).
+2. We also import less votes in total, as we avoid importing statements for
+   candidates that never got successfully backed on any chain.
+
+It also is secure, because disputes are only ever raised in the approval voting
+phase. A node only starts the approval process after it has seen a candidate
```

Comment:
I am personally missing some context and especially the sequence of the different phases referenced here, but assuming the intended audience would have gained more familiarity.

Reply:
Yep phases are explained in the remainder of the guide, so that should be good: But thanks for pointing that out - that's precisely the feedback I was hoping for. :-)